src/styles/main.scss

```diff
@@ -1,3 +1,54 @@
+@import "./utils/variables";
+@import "./utils/functions";
+@import "./utils/resets";
+
 body {
+  line-height: 100px;
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
   margin: 0;
 }
+
+.calendar {
+  display: flex;
+  margin: 150px auto;
+  flex-wrap: wrap;
+  gap: $gap;
+  width: width-calculate($day-size, $gap, $border-size);
+
+  @for $month-length from 28 through 31 {
+    &__month_length--#{$month-length} {
+      .calendar__day:nth-child(n + #{$month-length + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    height: $day-size;
+    width: $day-size;
+    box-sizing: border-box;
+    text-align: center;
+    background-color: #eee;
+    border: $border-size solid black;
+    transition: all 300ms ease-out;
+
+    @for $i from 1 through $month-length {
+      &:nth-child(#{$i})::before {
+        content: "#{$i}";
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffc0cb;
+      transform: translateY(-20px);
+    }
+
+    @each $theFirstDay, $numDay in $week {
+      &:first-child {
+        margin-left: 2 * ($day-size + $border-size);
+      }
+    }
```

Comment:
you should move this loop to your calendar class and create modifiers for each starting day, like if I use `calendar__start_day--mon` the first day should be on monday
![image](https://user-images.githubusercontent.com/37161315/216437482-c137c639-0b60-4909-8d64-a1262b755a8a.png)